pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -450,20 +451,26 @@ private void deleteSchemaInternal(String schemaName) {
     }
 
     // If the schema is associated with a table, we should not delete it.
-    List<String> tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
-    for (String tableName : tableNames) {
-      TableConfig config = _pinotHelixResourceManager.getRealtimeTableConfig(tableName);
-      String tableSchema = config.getValidationConfig().getSchemaName();
-
-      if (schemaName.equals(tableSchema)) {
+    // TODO: Check OFFLINE tables as well
```

Comment:
The main concern is that this is behavior change, and out of the scope of this PR. We can address this separately if desired, but need to explicitly call out this behavior change. Thus left a TODO so that people aware of this

Reply:
I think we should mention this in the TODO as well, why it was not implemented?